tools/wptrunner/wptrunner/testloader.py

```diff
@@ -29,29 +31,103 @@ def do_delayed_imports():
     from manifest.download import download_from_github  # type: ignore
 
 
-class TestGroupsFile:
-    """
-    Mapping object representing {group name: [test ids]}
-    """
-
-    def __init__(self, logger, path):
+class TestGroups:
+    def __init__(self, logger, path, subsuites):
         try:
             with open(path) as f:
-                self._data = json.load(f)
+                data = json.load(f)
         except ValueError:
             logger.critical("test groups file %s not valid json" % path)
             raise
 
+        self.tests_by_group = defaultdict(set)
         self.group_by_test = {}
-        for group, test_ids in self._data.items():
+        for group, test_ids in data.items():
             for test_id in test_ids:
-                self.group_by_test[test_id] = group
+                subsuite, group_name = group.split(":", 1)
+                if subsuite is None:
+                    group_name = subsuite
+                    subsuite = None
```

Comment:
Do not understand the intention here. Why we need the assignment at line 50 if `subsuite` is already None? Why we want to set `group_name` to None?